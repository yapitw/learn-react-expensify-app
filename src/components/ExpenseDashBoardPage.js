import React from 'react';

const ExpenseDashBoardPage = () => (
  <div>
    This is from dashboard component
  </div>
);

export default ExpenseDashBoardPage;