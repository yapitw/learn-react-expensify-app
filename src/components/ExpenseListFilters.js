import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate } from '../actions/filters'
import { DateRangePicker } from 'react-dates';

export class ExpenseListFilters extends React.Component {
  state = {
    calenderFocused: null,
  };
  onDatesChange = ({ startDate, endDate }) => {
    this.props.setStartDate(startDate);
    this.props.setEndDate(endDate);
  };
  onFocusChange = (calenderFocused) => {
    this.setState(() => ({ calenderFocused }));
  };
  onTextChange = (e) => {
    this.props.setTextFilter(e.target.value);
  };
  onSortChange = (e) => {
    if (e.target.value === 'amount') {
      this.props.sortByAmount();
    } else if (e.target.value === 'date') {
      this.props.sortByDate();
    }
  };
  render() {
    return (
      <div className="content-container">
        <div className="input-group">
          <div className="input-group__item">
            <input
              type="text"
              className="text-input"
              placeholder="Search expenses"
              value={this.props.filters.text}
              onChange={this.onTextChange}
            />
          </div>
          <div className="input-group__item">
            <select
              className="select"
              value={this.props.filters.sortBy}
              onChange={this.onSortChange}
            >
              <option value="date">Date</option>
              <option value="amount">Amount</option>
            </select>
          </div>
          <div className="input-group__item">
            <DateRangePicker
              startDate={this.props.filters.startDate}
              endDate={this.props.filters.endDate}
              onDatesChange={this.onDatesChange}
              focusedInput={this.state.calenderFocused}
              onFocusChange={this.onFocusChange}
              showClearDates={true}
              numberOfMonths={1}
              isOutsideRange={() => false}
            />
          </div>
        </div>
      </div>
    )
  }
}

const mapStateToProps = (state) => ({
  filters: state.filters
});

const mapDispatchToProps = (dispatch) => ({
  setStartDate: (startDate) => dispatch(setStartDate(startDate)),
  setEndDate: (endDate) => dispatch(setEndDate(endDate)),
  setTextFilter: (value) => dispatch(setTextFilter(value)),
  sortByAmount: () => dispatch(sortByAmount()),
  sortByDate: () => dispatch(sortByDate()),
});

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);