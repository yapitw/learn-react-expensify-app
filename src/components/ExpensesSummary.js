import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import expensesTotal from '../selectors/expenses-total';
import selectExpense from '../selectors/expenses';


export const ExpensesSummary = ({ expensesCount, expensesTotal }) => (
  <div>
    {expensesCount !== 0 && (
      <h3>
        Viewing {expensesCount} expense{expensesCount > 1 && 's'} totalling {numeral(expensesTotal / 100).format('$0,0.00')}
      </h3>
    )}
  </div>
);

const mapStateToProps = (state, props) => ({
  expenses: selectExpense(state.expenses, state.filters),
  expensesTotal: expensesTotal(state.expenses),
  expensesCount: state.expenses.length
});

export default connect(mapStateToProps)(ExpensesSummary);