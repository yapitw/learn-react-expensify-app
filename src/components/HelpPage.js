import React from "react";

const HelpPage = () => (
  <div>
    This is from my add help component
  </div>
);

export default HelpPage;