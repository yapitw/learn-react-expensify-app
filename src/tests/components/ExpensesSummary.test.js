import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';


test('should render ExpensesSummary correctly with single expense data', () => {
  const wrapper = shallow(<ExpensesSummary expensesCount="1" expensesTotal="9434" />);
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary correctly with multi expenses data', () => {
  const wrapper = shallow(<ExpensesSummary expensesCount="2" expensesTotal="9434" />);
  expect(wrapper).toMatchSnapshot();
});
