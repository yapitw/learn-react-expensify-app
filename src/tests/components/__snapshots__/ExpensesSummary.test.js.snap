// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary correctly with multi expenses data 1`] = `
<div>
  <h3>
    Viewing 
    2
     expense
    s
     totalling 
    $94.34
  </h3>
</div>
`;

exports[`should render ExpensesSummary correctly with single expense data 1`] = `
<div>
  <h3>
    Viewing 
    1
     expense
     totalling 
    $94.34
  </h3>
</div>
`;
